spring-cloud-gcp-starters/spring-cloud-gcp-starter-logging/src/main/java/org/springframework/cloud/gcp/logging/autoconfig/StackdriverLoggingProperties.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging.autoconfig;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Chengyuan Zhao
+ */
+@ConfigurationProperties("spring.cloud.gcp.logging.traceId")
+public class StackdriverLoggingProperties {
+	private TraceIdExtractorType extractorType;
+
+	public TraceIdExtractorType getExtractorType() {
```

Comment:
I think we wanted to call the property just "extractor" instead of "extractorType".

Reply:
I think extractorType is more appropriate though. In the settings class we have the getter and that getter's name makes more sense as `getExtractorType` vs `getExtractor`, which implies getting the actual extractor instead of the type.

Reply:
The properties bean is a special beast. It determines the configuration surface.
Basically we're choosing between:

`spring.cloud.gcp.logging.traceId.extractor=XCLOUD`
vs
`spring.cloud.gcp.logging.traceId.extractorType=XCLOUD`

If you both think `...extactorType` is better, I'm fine with that.

Reply:
Yeah... the `getExtractor()` is really confusing on the code level, but `extractor` is shorter for the configuration properties. The JavaDocs and return type does the trick. So, I'm really more up to the `extractor`, but can live with the `extactorType` since we have similar experience in Spring Boot already: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java#L511

Reply:
Sounds like it's an established pattern in Spring Boot.
@ChengyuanZhao let's change to `extractor`, and then we can merge.